src/once_cell.rs

```diff
@@ -0,0 +1,800 @@
+use std::cell::UnsafeCell;
+use std::convert::Infallible;
+use std::fmt;
+use std::future::Future;
+use std::mem::{forget, MaybeUninit};
+use std::pin::Pin;
+use std::ptr;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::task::{Context, Poll, RawWaker, RawWakerVTable, Waker};
+
+use event_listener::{Event, EventListener};
+use futures_lite::future;
+
+/// The `OnceCell` is uninitialized.
+const UNINITIALIZED: usize = 0;
+/// The `OnceCell` is being initialized.
+const INITIALIZING: usize = 1;
+/// The `OnceCell` is initialized.
+const INITIALIZED: usize = 2;
```

Comment:
this smells of Golang. :( Wouldn't an enum be better?

Reply:
It needs to fit into an `AtomicUsize`.

You know, this makes me wonder if a proc-macro type "atomic enum" crate wouldn't be a bad idea...

Reply:
Since it's private, we can use [enum w/o any additional deps](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=884e88217b8ee961a7dfe28d81781662).